apps/dh/api-dh/source/DataHub.WebApi/Controllers/MarketParticipantUserController.cs

```diff
@@ -51,17 +55,46 @@ public Task<ActionResult<GetAssociatedUserActorsResponseDto>> GetUserActorsAsync
         [Route("GetUserActorsByUserId")]
         public Task<ActionResult<GetAssociatedUserActorsResponseDto>> GetUserActorsByUserIdAsync(Guid userId)
         {
-            return HandleExceptionAsync(() => _client.GetUserActorsAsync(userId));
+            return HandleExceptionAsync(() => _marketParticipantClient.GetUserActorsAsync(userId));
         }
 
         /// <summary>
         ///     Retrieves the audit log history for the specified user.
         /// </summary>
         [HttpGet]
-        [Route("GetUserHistory")]
-        public Task<ActionResult<UserAuditLogsDto>> GetUserHistoryAsync(Guid userId)
+        [Route("GetUserAuditLogs")]
+        public Task<ActionResult<MarketParticipant.Dto.UserAuditLogsDto>> GetUserAuditLogsAsync(Guid userId)
         {
-            return HandleExceptionAsync(() => _client.GetUserAuditLogsAsync(userId));
+            return HandleExceptionAsync(async () =>
+            {
+                var auditLogs = await _marketParticipantClient
+                    .GetUserAuditLogsAsync(userId)
+                    .ConfigureAwait(false);
+
+                var roleAssignmentAuditLogs = new List<UserRoleAssignmentAuditLogDto>();
+
+                foreach (var auditLog in auditLogs.UserRoleAssignmentAuditLogs)
+                {
+                    var userDto = await _marketParticipantClient
```

Comment:
We should consider making a function that can fetch multiple user in a single call

Reply:
In case of perfromance issues, we should add query APIs for this, yes. But in this case, the list will always be short, so I see no reason to optimise at the moment.

Reply:
Ok :)